radis/lbl/factory.py

```diff
@@ -1542,6 +1575,77 @@ def _generate_wavenumber_arrays(self):
 
         return
 
+    def predict_time(self):
+        wstep = self.params.wstep
+        broadening_max_width = self.params.broadening_max_width
+        n_lines = self.misc.total_lines
+        spectral_points = (
+            self.params.wavenum_max_calc - self.params.wavenum_min_calc
+        ) / self.params.wstep
+
+        df = self.df0.copy()
+
+        from radis.lbl.broadening import get_molar_mass, voigt_broadening_HWHM
+
+        # Defining parameters
+        Tgas = 300
+        pressure_mbar = self.input.pressure_mbar
+        mole_fraction = self.input.mole_fraction
+        # convert from mbar to atm for linebroadening calculation
+        pressure_atm = pressure_mbar / 1013.25
+        # coefficients tabulation temperature
+        Tref = self.input.Tref
+        # Check self broadening is here
+        if not "Tdpsel" in list(df.keys()):
+            Tdpsel = None  # if None, voigt_broadening_HWHM uses df.Tdpair
+        else:
+            Tdpsel = df.Tdpsel
+
+        molar_mass = get_molar_mass(df)
+
+        # Calculate broadening FWHM
+        _, wl, wg = voigt_broadening_HWHM(
```

Comment:
I'd remove all calculation from predict_time()  ; it makes things complicated. 

We can call it during the calculation ; once we know most of the parameters. If some parmaeters are not known when it is called ; just raise an error, or add a default value.  Let's not mix calculation & the "package" around, i.e. prediction and analysis of results. 

 For instance, do not re-init the w_axis grids .  You can call predict() during the calculation once you know all the parameters; for instance replacing the ["Calculating. X lines expect Y s".](https://github.com/radis/radis/blob/c18b53522c2ca1b6486d6f7828aa108f58691639/radis/lbl/broadening.py#L2225)